lib/cc/yaml/nodes/root.rb

```diff
@@ -16,6 +20,17 @@ def nested_warnings(*)
         def inspect
           "#<#{self.class.name}:#{super}>"
         end
+
+        def exclude?(path)
+          exclude_paths.present? &&
+            exclude_paths.match_any?(path)
+        end
+
+        def rate?(path)
+          ratings.nil? ||
+            ratings.paths.blank? ||
```

Comment:
I'm concerned `Root` will become a God class (or at least a God delegator), but fine for now and we can see how it evolves.